packages/wasm-dpp/src/state_repository.rs

```diff
@@ -480,23 +502,31 @@ impl StateRepositoryLike for ExternalStateRepositoryLikeWrapper {
         todo!()
     }
 
-    async fn fetch_latest_platform_block_header(&self) -> anyhow::Result<Vec<u8>> {
-        let header = self
+    async fn fetch_latest_platform_block_header(&self) -> Result<Vec<u8>> {
+        let value: JsValue = self
             .0
-            .lock()
-            .expect("unexpected concurrency issue!")
-            .fetch_latest_platform_block_header();
-        Ok(header)
+            .fetch_latest_platform_block_header()
+            .await
+            .map_err(from_js_error)?;
+
+        Ok(Uint8Array::new(&value).to_vec())
     }
 
-    async fn fetch_latest_platform_core_chain_locked_height(&self) -> anyhow::Result<Option<u32>> {
-        let height = self
+    async fn fetch_latest_platform_core_chain_locked_height(&self) -> Result<Option<u32>> {
+        let maybe_height = self
             .0
-            .lock()
-            .expect("unexpected concurrency issue!")
             .fetch_latest_platform_core_chain_locked_height()
-            .map(Into::into);
-        Ok(height)
+            .await
+            .map_err(from_js_error)?;
+
+        if maybe_height.is_falsy() {
```

Comment:
Let's check that it's undefined

Reply:
Fixed. Although I had to leave `is_falsy()` check in FetchTransactionResponse because this is what one of the tests expect to happen